src/NzbDrone.Core/HealthCheck/Checks/MountCheck.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Linq;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Tv;
+
+namespace NzbDrone.Core.HealthCheck.Checks
+{
+    public class MountCheck : HealthCheckBase
+    {
+        private readonly IDiskProvider _diskProvider;
+        private readonly ISeriesService _seriesService;
+
+        public MountCheck(IDiskProvider diskProvider, ISeriesService seriesService)
+        {
+            _diskProvider = diskProvider;
+            _seriesService = seriesService;
+        }
+
+        public override HealthCheck Check()
+        {
+            var mounts = _seriesService.GetAllSeries()
+                                       .Select(series => _diskProvider.GetMount(series.Path))
+                                       .DistinctBy(m => m.RootDirectory)
```

Comment:
Move DistinctBy one line up: `.DistinctBy(m => m.Path)`. That way you don't call GetMount per series, but only per Path.